waspc/src/Wasp/ExternalConfig/TsConfig.hs

```diff
@@ -0,0 +1,134 @@
+      <*> v .:? "lib"
+      <*> v .:? "allowJs"
+      <*> v .:? "typeRoots"
+      <*> v .:? "outDir"
+
+analyzeTsConfigContent :: Path' Abs (Dir WaspProjectDir) -> IO (Either [CompileError] TsConfig)
+analyzeTsConfigContent waspDir = runExceptT $ do
+  tsConfigFile <- ExceptT findTsConfigOrError
+  tsConfig <- ExceptT $ readTsConfigFile tsConfigFile
+  ExceptT $ validateTsConfig tsConfig
+  where
+    findTsConfigOrError = maybeToEither [fileNotFoundMessage] <$> findTsConfigFile waspDir
+    fileNotFoundMessage = "Couldn't find the tsconfig.json file in the " ++ toFilePath waspDir ++ " directory"
+
+findTsConfigFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
+findTsConfigFile waspProjectDir = findFileInWaspProjectDir waspProjectDir tsConfigInWaspProjectDir
+
+readTsConfigFile :: Path' Abs File' -> IO (Either [CompileError] TsConfig)
+readTsConfigFile tsConfigFile = do
+  tsConfigContent <- IOUtil.readFileBytes tsConfigFile
+
+  parseResult <- parseJsonWithComments . BS.toString $ tsConfigContent
+
+  case parseResult of
+    Right tsConfig -> return $ Right tsConfig
+    Left err -> return $ Left ["Failed to parse tsconfig.json file: " ++ err]
+
+validateTsConfig :: TsConfig -> IO (Either [CompileError] TsConfig)
+validateTsConfig tsConfig =
+  return $
+    if null tsConfigErrors
+      then Right tsConfig
+      else Left tsConfigErrors
+  where
+    tsConfigErrors =
+      concat
+        [ validateRequiredFieldInCompilerOptions "module" "esnext" _module,
+          validateRequiredFieldInCompilerOptions "target" "esnext" target,
+          validateRequiredFieldInCompilerOptions "moduleResolution" "bundler" moduleResolution,
+          validateRequiredFieldInCompilerOptions "jsx" "preserve" jsx,
+          validateRequiredFieldInCompilerOptions "strict" True strict,
+          validateRequiredFieldInCompilerOptions "esModuleInterop" True esModuleInterop,
+          validateRequiredFieldInCompilerOptions "lib" ["dom", "dom.iterable", "esnext"] lib,
+          validateRequiredFieldInCompilerOptions "allowJs" True allowJs,
+          validateRequiredFieldInCompilerOptions "typeRoots" ["node_modules/@testing-library", "node_modules/@types"] typeRoots,
+          validateRequiredFieldInCompilerOptions "outDir" ".wasp/phantom" outDir
+        ]
+
+    validateRequiredFieldInCompilerOptions fieldName expectedValue getField =
+      validateRequiredField ("compilerOptions." ++ fieldName) expectedValue $ getField compilerOptionsFields
+
+    compilerOptionsFields = compilerOptions tsConfig
+
+-- | Used to show expected values in error messages.
+class ShowJs a where
+  showJs :: a -> String
+
+instance ShowJs String where
+  showJs = show
+
+instance ShowJs [String] where
+  showJs = show
+
+instance ShowJs Bool where
+  showJs True = "true"
+  showJs False = "false"
+
+type FieldName = String
+
+validateRequiredField :: (Eq value, ShowJs value) => FieldName -> value -> Maybe value -> [CompileError]
```

Comment:
I mentioned this in previous PRs already, taking `Maybe` as an arugment is always a bit suspicious. You should probably rather instead handle that outside, and take just `value` here.